### ISSUE TITLE
Several edge types in shortestpath()

### ISSUE THREAD

**Issue by dargolith** [orientechnologies/orientdb#5711]
Hi,

It would be nice to specify more than one allowed edge type for shortestpath. It would mean: "Give me the shortest path only using one of these edge classes".

Is there a reason for it not being supported?

I only briefly glanced at the code, but it seems to me like the inner algorithm itself already supports it.

```
String   edgeType;
String[] edgeTypeParam;

...

ctx.edgeType = null;
if (iParams.length > 3) {
    ctx.edgeType = iParams[3] == null ? null : "" + iParams[3];
}
ctx.edgeTypeParam = new String[] { ctx.edgeType };
```

Seems like `ctx.edgeTypeParam` is the one being used later in the code and it is an array already. Would it work by just allowing an array argument for the function?

**Comment by katherinessmith** [comment]
https://github.com/orientechnologies/orientdb/issues/7842 requested the same change. Fix is in https://github.com/orientechnologies/orientdb/commit/3b660af55ee8f6fa34336a47fde71c9e93504175 and will be released in 2.2.30.

**Comment by luigidellaquila** [comment]
Thank you very much @katherinessmith 

I think we can close this

Thanks

Luigi
